support-frontend/assets/components/subscriptionCheckouts/paymentMethodSelector.tsx

```diff
@@ -6,68 +6,99 @@ import {
 	SvgDirectDebit,
 	SvgPayPal,
 } from '@guardian/source-react-components';
-import type { Node } from 'react';
+import type { ReactNode } from 'react';
 import Rows from 'components/base/rows';
-import 'helpers/types/option';
 import type { PaymentMethod } from 'helpers/forms/paymentMethods';
-import 'helpers/forms/paymentMethods';
 import type { Option } from 'helpers/types/option';
 
 type PropTypes = {
 	availablePaymentMethods: PaymentMethod[];
 	paymentMethod: Option<PaymentMethod>;
-	setPaymentMethod: (...args: any[]) => any;
+	setPaymentMethod: (method: PaymentMethod) => void;
 	validationError: string | undefined;
 };
+
 type RadioWithImagePropTypes = {
 	id: string;
-	image: Node;
+	image: ReactNode;
 	label: string;
 	name: string;
 	checked: boolean;
-	onChange: (...args: any[]) => any;
+	onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
 };
+
 const radioWithImageStyles = css`
 	display: inline-flex;
 	justify-content: space-between;
 	align-items: center;
 `;
+
 const paymentIcon = css`
 	min-width: 30px;
 	max-width: 40px;
 `;
 
-function RadioWithImage(props: RadioWithImagePropTypes) {
+function RadioWithImage({
+	id,
+	image,
+	label,
+	checked,
+	name,
+	onChange,
+}: RadioWithImagePropTypes) {
 	return (
 		<div css={radioWithImageStyles}>
-			<Radio {...props} />
-			<div css={paymentIcon}>{props.image}</div>
+			<Radio
+				id={id}
+				label={label}
+				checked={checked}
+				name={name}
+				onChange={onChange}
+			/>
+			<div css={paymentIcon}>{image}</div>
 		</div>
 	);
 }
 
-const paymentMethodIcons: Record<PaymentMethod, Node> = {
+const paymentMethodIcons: Record<PaymentMethod, ReactNode> = {
 	Stripe: <SvgCreditCard />,
 	PayPal: <SvgPayPal />,
 	DirectDebit: <SvgDirectDebit />,
+	Sepa: '',
+	ExistingCard: '',
+	ExistingDirectDebit: '',
+	AmazonPay: '',
+	None: '',
```

Comment:
I've added the other `PaymentMethod` fields to these types as defining a more limited type for 'payment methods we only use on subscriptions' caused problems with the way this component is used. In practice we only ever pass in the first three as allowed payment methods.

Reply:
If we narrow the return type for the [supportedPaymentMethods function](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/helpers/subscriptionsForms/countryPaymentMethods.ts#L9) I think we can avoid having to add in these extra fields

Reply:
You're right, I think that would be better 👍